packages/react/src/shared/types.ts

```diff
@@ -0,0 +1,27 @@
+import { QueryClient } from '@tanstack/react-query';
+import { AnyRouter, MaybePromise } from '@trpc/server';
+
+/**
+ * @internal
+ */
+export interface UseMutationOverride {
+  onSuccess: (opts: {
+    /**
+     * Calls the original function that was defined in the query's `onSuccess` option
+     */
+    originalFn: () => MaybePromise<unknown>;
+    queryClient: QueryClient;
+  }) => MaybePromise<unknown>;
+}
+
+/**
+ * @internal
+ */
+export interface CreateTRPCReactOptions<_TRouter extends AnyRouter> {
+  /**
+   * Override behaviors of the built-in hooks
+   */
+  unstable_overrides?: {
+    useMutation?: Partial<UseMutationOverride>;
+  };
```

Comment:
allow overriding of other hooks too?

Reply:
not sure what the use case of overriding useQuery might be though.

Reply:
Meh. Maybe later! :D